Small update to Type checks and casts page (#3710)
JetBrains/kotlin-web-site: docs/topics/typecasts.md

[//]: # (title: Type checks and casts)

> In Kotlin, you can perform type checks to check what type an object is at runtime. Type casts allow you to convert objects

Reviewer: ```suggestion
In Kotlin, you can perform type checks to check the type of an object at runtime. Type casts convert objects
```

}
```

> Note that smart casts work only when the compiler can guarantee that the variable won't change between the check and its usage.

Reviewer: I think an obligatory requirement like that should be an actual note (warning) or have a separate header

<table header-style="none">
        <tr>
        <td>
            <code>val</code> local variables
        </td>
        <td>
            Always, except <a href="delegated-properties.md">local delegated properties</a>.
        </td>
    </tr>
        <tr>
        <td>
            <code>val</code> properties
        </td>
        <td>
            If the property is <code>private</code>, <code>internal</code>, or if the check is performed in the same <a href="visibility-modifiers.md#modules">module</a> where the property is declared. Smart casts can't be used on <code>open</code> properties or properties that have custom getters.
        </td>
    </tr>
        <tr>
        <td>
            <code>var</code> local variables
        </td>
        <td>
            If the variable is not modified between the check and its usage, is not captured in a lambda that modifies it, and is not a local delegated property.
        </td>
    </tr>
        <tr>
        <td>
            <code>var</code> properties
        </td>
        <td>
            Never, because the variable can be modified at any time by other code.
        </td>
    </tr>
</table>

## "Unsafe" cast operator

> Usually, the cast operator throws an exception if the cast isn't possible. Thus, it's called *unsafe*.

Reviewer: ```suggestion
Usually, the cast operator throws an exception if the cast isn't possible. Thus, it's called _unsafe_.
```
